src/ucp/wireup/wireup.c

```diff
@@ -1666,6 +1760,7 @@ ucs_status_t ucp_wireup_init_lanes(ucp_ep_h ep, unsigned ep_init_flags,
                             UCP_EP_FLAG_REMOTE_CONNECTED);
     }
 
+    ucs_queue_head_init(&replay_pending_queue);
```

Comment:
why needed? shall we remove the queue initialization from ucp_wireup_eps_pending_extract?

Reply:
it's required because `ucp_wireup_eps_pending_extract` is called after `if (ucp_ep->cfg_index == UCP_WORKER_CFG_INDEX_NULL)`, but the queue needs to be initialized also when `cfg_index == UCP_WORKER_CFG_INDEX_NULL`

Reply:
and we cannot remove queue initialization from `ucp_wireup_eps_pending_extract` because it is being reused in `wireup_cm`

Reply:
I see the queue is initialized before the cfg_index check:
https://github.com/openucx/ucx/blob/fa0fbdbf8288e30dae53bfaed1c8bc760b9795a9/src/ucp/wireup/wireup_ep.c#L656-L661

Reply:
I was refering to this:
https://github.com/openucx/ucx/blob/112084e6bf4a1ece0d4faec81f4dc155b370b376/src/ucp/wireup/wireup.c#L1716